src/workflows-app/SubmitWorkflow.js

```diff
@@ -91,6 +91,7 @@ export const SubmitWorkflow = wrapWorkflowsPage({ name: 'SubmitWorkflow' })(
 
         if (cbasProxyUrlState.status === AppProxyUrlStatus.Ready) {
           await loadRunsData(cbasProxyUrlState);
+          await refreshApps(true);
```

Comment:
both changes to this file are piggyback changes based on some behaviors I noticed while testing the button changes. it should smooth out some rendering weirdness in the UI.